src/Storages/VirtualColumnUtils.cpp

```diff
@@ -156,8 +156,13 @@ bool prepareFilterBlockWithQuery(const ASTPtr & query, ContextPtr context, Block
         auto actions = std::make_shared<ActionsDAG>(block.getColumnsWithTypeAndName());
         PreparedSets prepared_sets;
         SubqueriesForSets subqueries_for_sets;
+        const NamesAndTypesList source_columns;
+        const NamesAndTypesList aggregation_keys;
+        const ColumnNumbersList grouping_set_keys;
+
         ActionsVisitor::Data visitor_data(
-            context, SizeLimits{}, 1, {}, std::move(actions), prepared_sets, subqueries_for_sets, true, true, true, false);
+            context, SizeLimits{}, 1, source_columns, std::move(actions), prepared_sets, subqueries_for_sets, true, true, true, false,
```

Comment:
Here we for more than a year had a potential undefined behavior. `ActionsVisitor::Data` stores const reference to `source_columns` but temporary object was provided to the constructor.
I explicitly restricted passing of temporary objects to `ActionsVisitor::Data` constructor.